### DISCUSSION TITLE
API router throwing error with large files

### DISCUSSION THREAD

**Discussion by mihail727** [nuxt/nuxt#16353] [Questions]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.0`
- Nuxt Version:     `3.0.0-rc.0-27508091.78fcbcf`
- Package Manager:  `yarn@1.22.18`
- Builder:          `vite`
- User Config:      `runtimeConfig`, `buildModules`, `modules`, `tailwindcss`, `app`, `nitro`, `build`, `modern`
- Runtime Modules:  `@nuxtjs/tailwindcss@5.0.3`
- Build Modules:    `@vueuse/nuxt@8.3.1`
------------------------------

### Reproduction

https://stackblitz.com/edit/github-tjflbh?file=app.vue

Reproduction does not allow opening files
Therefore, there is only a structure

### Describe the bug

When i'm trying to upload from client to server the large file - i'm getting error:
```
ERROR  [h3] write EPIPE
at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:94:16)
```

**Comment by mihail727** [comment]
I managed to start uploading large files by using `multer` and `callHanlder`:
`@/server/api/upload.ts`
```ts
import { CompatibilityEvent, callHandler } from 'h3';
import multer from 'multer';

const storage = multer.diskStorage({
	destination: (req, file, cb) => {
		cb(null, './');
	},
	filename: (req, file, cb) => {
		cb(null, file.originalname);
	},
});

export default async (event: CompatibilityEvent): Promise<IServerResponse> => {
	await callHandler(multer({ storage }).single('file'), event.req, event.res);
	return { success: true };
};
```
It turns out all this time it worked just Typescript complains about types:
![](https://i.imgur.com/RsqKN8K.png)
text:
```
An argument of type "RequestHandler<ParamsDictionary, any, any, ParsedQs, Record<string, any>>" cannot be assigned to a parameter of type "Middleware".
   Parameter types "req" and "req" are incompatible.
     The 'IncomingMessage' type is missing the following properties from the 'Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>' type: get, header, accepts, acceptsCharsets and 25.ts(2345) more
```

**Reply by jeffpohlmeyer** [reply]
@mihail727 the `callHandler` method is not currently a named export from 'h3'. How would you go about doing this without that method?

**Comment by mihail727** [comment]
@danielroe i'm sorry but why did you move this to the discussion?

**Reply by danielroe** [reply]
It seemed more like a 'how do I do this' question rather than an issue or bug in Nuxt itself. Particularly as you have a solution, but others may also have alternative suggestions. Do you believe there is an issue with Nuxt?

**Note**: the type discrepancy is likely not something we can fix on the Nuxt side. The [multer types](https://unpkg.com/browse/@types/multer@1.4.7/index.d.ts) are dependent on express itself.

**Reply by mihail727** [reply]
> 

I just hoped that you could remove this warning from the nuxt side
